### PR TITLE
Videos and documents

### PR DESCRIPTION
**Pull request by notetell**

### CODE REVIEW

File: lib/kanal/interfaces/telegram/telegram_interface.rb [hunk 1]
```diff
@@ -20,67 +20,46 @@ def initialize(core, bot_token)
 
           @core.register_plugin Kanal::Plugins::Batteries::BatteriesPlugin.new
           @core.register_plugin Kanal::Interfaces::Telegram::Plugins::TelegramIntegrationPlugin.new
+
         end
 
         def start
           ::Telegram::Bot::Client.run(@bot_token) do |bot|
             bot.listen do |message|
               puts "message class: #{message.class}"
```
Comment (idchlife) [inline lib/kanal/interfaces/telegram/telegram_interface.rb:29]: Remove puts and replace them with logging if needed

**Review by idchlife — CHANGES_REQUESTED** [review]

File: .idea/kanal-interfaces-telegram.iml
[the remark this answers is not in the document]
Comment (notetell) [reply .idea/kanal-interfaces-telegram.iml:1]: Done

File: lib/kanal/interfaces/telegram/plugins/telegram_integration_plugin.rb [hunk 2]
```diff
@@ -16,38 +16,30 @@ def setup(core)
...
 
           def register_parameters(core)
-            core.register_input_parameter :tg_message, readonly: true
-            core.register_input_parameter :tg_text, readonly: true
-            core.register_input_parameter :tg_callback, readonly: true
-            core.register_input_parameter :tg_callback_text, readonly: true
             core.register_input_parameter :tg_chat_id, readonly: true
             core.register_input_parameter :tg_username, readonly: true
+            core.register_input_parameter :tg_text, readonly: true
+            core.register_input_parameter :tg_button, readonly: true
```
Comment (idchlife) [inline lib/kanal/interfaces/telegram/plugins/telegram_integration_plugin.rb:22]: Here you define :tg_button, but at runtime you use :tg_button_pressed

If this is indeed error - rewrite interface so you can test it in spec with fake telegram message object filled with parameters.

Move processing of message from bot.listen do |message| to separate method and remove explicit check for types but check for specific methods instead (or check if !something.nil?) so method can be tested in spec file.

Processing of telegram message and creating kanal input out of it - needs testin in spec file

**Review by idchlife — CHANGES_REQUESTED** [review]

File: lib/kanal/interfaces/telegram/plugins/telegram_integration_plugin.rb (hunk 2 above)
Comment (notetell) [reply lib/kanal/interfaces/telegram/plugins/telegram_integration_plugin.rb]: Reverted parameter name change, moved processing to separate method, test for this method created https://github.com/idchlife/kanal-interfaces-telegram/pull/3/commits/cfb7fcc6533ec22a2e882d1b5443a9ea5a8f39bb

File: lib/kanal/helpers/telegram_link_parser.rb
Comment (idchlife) [inline lib/kanal/helpers/telegram_link_parser.rb:1]: This helper in Kanal namespace, but it is used and intended for usage only in telegram interface.

**Review by idchlife — CHANGES_REQUESTED** [review]

File: lib/kanal/helpers/telegram_link_parser.rb
Comment (notetell) [reply lib/kanal/helpers/telegram_link_parser.rb:1]: Moved in Telegram namespace https://github.com/idchlife/kanal-interfaces-telegram/pull/3/commits/3aa3b939bf77e6b55d5b74a703dc6e2762de1980

**Review by idchlife — APPROVED** [review]

**Pull request merged**